reconcile/aus/advanced_upgrade_service.py

```diff
@@ -1,25 +1,334 @@
+        org=AUSOCMOrganization(
+            **data_default_none(
+                AUSOCMOrganization,
+                dict(
+                    name=org_id,
+                    orgId=org_id,
+                    blockedVersions=org_labelset.blocked_versions,
+                    environment=ocm_env,
+                    addonManagedUpgrades=False,
+                    sectors=org_labelset.sector_dependencies(),
+                ),
+            )
+        )
+    )
+
+    # init policy for each cluster
+    for c in clusters:
+        try:
+            upgrade_policy = _build_policy_from_labels(c.labels)
+            org_upgrade_spec.specs.append(
+                ClusterUpgradeSpec(
+                    name=c.ocm_cluster.name,
+                    cluster_uuid=c.ocm_cluster.external_id,
+                    ocm=org_upgrade_spec.org,
+                    upgradePolicy=upgrade_policy,
+                )
+            )
+        except ValidationError as validation_error:
+            for e in validation_error.errors():
+                org_upgrade_spec.add_cluster_error(
+                    c.ocm_cluster.external_id, f"label {e['loc'][0]}: {e['msg']}"
+                )
+
+    return org_upgrade_spec
+
+
+class ClusterUpgradePolicyLabelSet(BaseModel):
+    """
+    Parses, represents and validates a set of subscription labels for AUS.
+    """
+
+    soak_days: int = Field(alias=aus_label_key("soak-days"), ge=0)
+    workloads: CSV = Field(alias=aus_label_key("workloads"), csv_min_items=1)
+    schedule: str = Field(alias=aus_label_key("schedule"))
+    mutexes: Optional[CSV] = Field(alias=aus_label_key("mutexes"))
+    sector: Optional[str] = Field(alias=aus_label_key("sector"))
+    _schedule_validator = validator("schedule", allow_reuse=True)(cron_validator)
+
+
+def _build_policy_from_labels(labels: LabelContainer) -> ClusterUpgradePolicy:
+    """
+    Build a cluster upgrade policy object from a set of OCM labels. Parsing
+    and validation of the labels is delegated to the pydantic dataclass
+    ClusterUpgradePolicyLabelSet.
+    """
+    policy_labelset = build_labelset(labels, ClusterUpgradePolicyLabelSet)
+    return ClusterUpgradePolicy(
+        workloads=policy_labelset.workloads,
+        schedule=policy_labelset.schedule,
+        conditions=ClusterUpgradePolicyConditionsV1(
+            soakDays=policy_labelset.soak_days,
+            mutexes=policy_labelset.mutexes,
+            sector=policy_labelset.sector,
+        ),
+    )
+
+
+#
+# Feedback mechanism
+#
+
+
+def _signal_validation_issues_for_org(
+    ocm_api: OCMBaseClient,
+    org_upgrade_spec: OrganizationUpgradeSpec,
+) -> None:
+    """
+    Signal the validation errors of an organization to the users.
+    Right now it uses OCM service logs, but it could be extended to use
+    other mechanisms like slack etc.
+    """
+    org_id = org_upgrade_spec.org.org_id
+    ocm_env_name = org_upgrade_spec.org.environment.name
+    logging.error(
```

Comment:
I assume these are user errors, do we really care about these?

Reply:
I think a different log level could make sense.

Reply:
changed to warning